bigquery-storage/api/BigQueryStorage.Samples/Quickstart.cs

```diff
@@ -0,0 +1,90 @@
+﻿/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// [START bigquerystorage_quickstart]
+
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using Google.Cloud.BigQuery.Storage.V1;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using static Google.Cloud.BigQuery.Storage.V1.ReadSession.Types;
+
+public class QuickstartSample
+{
+    public struct Data
+    {
+        public string Name { get; set; }
+        public string State { get; set; }
+        public long Number { get; set; }
+    }
+
+    public async Task<List<Data>> QuickstartAsync(string projectId)
+    {
+        BigQueryReadClient bigQueryReadClient = BigQueryReadClient.Create();
+
+        // This example uses baby name data from the public datasets.
+        var srcTable = "projects/bigquery-public-data/datasets/usa_names/tables/usa_1910_current";
```

Comment:
nit: Given these variables are used only once, we can possibly inline them. I am fine either ways.